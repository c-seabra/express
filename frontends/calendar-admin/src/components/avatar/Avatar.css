.avatar {
  width: 100%;
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: space-between;
}

.avatarImg {
  position: relative;
}

.avatarImg object,
.avatarImg img {
  width: 100%;
  max-width: 42px;
  max-height: 42px;
  border-radius: 50%;
  border: 2px solid transparent;
}

.avatarImg svg {
  position: absolute;
  bottom: 2px;
  right: 2px;
  width: 16px;
  height: 16px;
}

.content {
  width: calc(100% - 72px)
}

.contentTitle {
  margin: 0;
}

.contentSubtitle {
  margin: .275rem 0 0;
  font-size: .875rem;
}

.icon {
  width: 20px;
}
