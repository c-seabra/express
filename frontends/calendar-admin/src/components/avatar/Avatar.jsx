import React from 'react'

import Accepted from '../svgs/Accepted'
import AvatarIcon from '../svgs/AvatarIcon'
import Declined from '../svgs/Declined'
import Tentative from '../svgs/Tentative'
import styles from './Avatar.css'

const Avatar = ({
  title,
  subTitle,
  src,
  alt,
  avatarType,
  response,
  invitationId,
  organizer,
  iconActive,
  listType,
  iconClickCallback,
  defaultSrc = 'https://s3-eu-west-1.amazonaws.com/assets.cilabs.com/images/person.svg'
}) => {
  // Will be updated with coresponding types/colors once we get color codes for different avatar types
  const avatarColor = avatarType === 'attendee' ? 'red' : 'transparent'

  let responseElem
  switch (response) {
    case 'accepted':
      responseElem = <Accepted />
      break
    case 'declined':
      responseElem = <Declined />
      break
    default:
      responseElem = <Tentative />
      break
  }

  const avatarSubTitle = subTitle || organizer ? subTitle || 'Organizer' : ''

  return (
    <div className={styles.avatar}>
      <div className={styles.avatarImg}>
        <object
          data={src || defaultSrc}
          type='image/png'
        >
          <img src={defaultSrc} alt={alt} style={{ borderColor: avatarColor }} />
        </object>
        {responseElem}
      </div>

      <div className={styles.content}>
        {title && <p className={styles.contentTitle}>{title}</p>}
        {avatarSubTitle && <p className={styles.contentSubtitle}>{avatarSubTitle}</p>}
      </div>
      {!organizer && iconActive && (
        <span className={styles.icon} onClick={() => iconClickCallback(invitationId)}>
          <AvatarIcon iconType={listType} />
        </span>
      )}
    </div>
  )
}

export default Avatar
