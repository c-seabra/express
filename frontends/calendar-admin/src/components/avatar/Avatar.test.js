import React from 'react';
import renderer from 'react-test-renderer';
import Avatar from './Avatar.jsx';

describe('Avatar', () => {

    test('renders without props', () => {
        const component = renderer.create(
            <Avatar />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('renders with props', () => {
        const component = renderer.create(
            <Avatar
                src="https://s3-eu-west-1.amazonaws.com/avenger.cilabs.net/staging/avatars/medium/8a86c417e1f0bf13bcd3f1b33fe7f36dd29dc5ff.png?1524668194"
                defaultSrc="https://websummit.com/wp-content/uploads/2019/06/ws-logo-pride2.png"
                alt="Invitee avatar"
            />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

});
