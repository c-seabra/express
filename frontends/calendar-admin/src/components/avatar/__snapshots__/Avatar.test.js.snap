// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders with props 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <object
      data="https://s3-eu-west-1.amazonaws.com/avenger.cilabs.net/staging/avatars/medium/8a86c417e1f0bf13bcd3f1b33fe7f36dd29dc5ff.png?1524668194"
      type="image/png"
    >
      <img
        alt="Invitee avatar"
        src="https://s3-eu-west-1.amazonaws.com/assets.cilabs.com/images/person.svg"
        style={
          Object {
            "borderColor": "transparent",
          }
        }
      />
    </object>
    <svg
      viewBox="0 0 9 9"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <g>
          <g>
            <path
              d="M4.5 9a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z"
              fill="#757575"
            />
          </g>
        </g>
        <g>
          <path
            d="M2 4h5v1H2z"
            fill="#fff"
          />
        </g>
      </g>
    </svg>
  </div>
  <div
    className={undefined}
  />
</div>
`;

exports[`Avatar renders without props 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <object
      data={undefined}
      type="image/png"
    >
      <img
        alt={undefined}
        src="https://s3-eu-west-1.amazonaws.com/assets.cilabs.com/images/person.svg"
        style={
          Object {
            "borderColor": "transparent",
          }
        }
      />
    </object>
    <svg
      viewBox="0 0 9 9"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <g>
          <g>
            <path
              d="M4.5 9a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z"
              fill="#757575"
            />
          </g>
        </g>
        <g>
          <path
            d="M2 4h5v1H2z"
            fill="#fff"
          />
        </g>
      </g>
    </svg>
  </div>
  <div
    className={undefined}
  />
</div>
`;
